app/views/incomeSources/manage/ManageObligations.scala.html

```diff
@@ -0,0 +1,204 @@
+
+@import implicitDateFormatter.longDate
+
+@getMessage(key: String, args: String*) = @{
+    messages(s"business-added.$key", args: _*)
+}
+
+@getUniqueMessage(key: String, args: String*) = @{
+    messages(s"incomeSources.add.manageObligations.$key", args: _*)
+}
+
+@getBusinessName(name: String, mode: String) = @{
+    if(name == "Not Found") getUniqueMessage("sole-trader")
+    else name
+}
+
+@getPeriodMessage(changeTo: String) = @{
+    if (changeTo == "annual") getUniqueMessage("annually")
+    else getUniqueMessage("quarterly")
+}
+
+@getTaxYears(taxYear:String) = @{
+    taxYear.subSequence(0,4) + " " + getUniqueMessage("to") + " " + taxYear.subSequence(5,9)
+}
+
+@getPageTitle = @{
+    val title = getUniqueMessage("h1")+" "+getBusinessName(businessName, mode)+" "+getUniqueMessage("h2")+" "+getPeriodMessage(changeTo)+" "+getUniqueMessage("tax-year")+" "+getTaxYears(taxYear)
+    title.trim
+}
+
+@mainTemplate(pageTitle = getPageTitle,
+    backUrl = Some(backUrl),
+    isAgent = isAgent,
+    btaNavPartial = user.btaNavPartial) {
+
+    @formWithCSRF(action = postAction) {
+
+        <div class="govuk-panel govuk-panel--confirmation" id="banner">
+            <div class="govuk-panel__body govuk-!-margin-bottom-3">
+                @getUniqueMessage("h1")
+            </div>
+            <h1 class="govuk-panel__title">
+                @getBusinessName(businessName, mode)
+            </h1>
+            <div class="govuk-panel__body">
+                @getUniqueMessage("h2")
+                @getPeriodMessage(changeTo)
+                @getUniqueMessage("tax-year")
+                @getTaxYears(taxYear)
+            </div>
+        </div>
+
+        <h2 class="govuk-label-wrapper">
+            <label class="govuk-label govuk-label--l govuk-!-margin-top-8">
+                @getUniqueMessage("heading")
+            </label>
+        </h2>
+
+        @if(sources.quarterlyObligationsDatesYearOne.nonEmpty) {
+            <div class="box-simple" id="quarterly">
+                <div class="page_headers govuk-heading-m"> @getMessage("quarterly-heading") </div>
+                <p class="govuk-body"> @getMessage("quarterly-p1") </p>
+
+                <table class="govuk-table">
+                    <thead class="govuk-table__head">
+                        <tr class="govuk-table__row">
+                            <th scope="col" class="govuk-table__header">@(getMessage("table.heading-1") + " " + sources.quarterlyObligationsDatesYearOne.head.inboundCorrespondenceFrom.getYear + " " + getMessage("to") + " " + (sources.quarterlyObligationsDatesYearOne.head.inboundCorrespondenceFrom.getYear + 1))</th>
+                            <th scope="col" class="govuk-table__header">@getMessage("table.heading-2")</th>
+                        </tr>
+                    </thead>
+
+                    <tbody class="govuk-table__body">
+                    @sources.quarterlyObligationsDatesYearOne.map { x =>
```

Comment:
preferable to use .foreach over .map as there is not return value here, but we evaluating view/render final HTML here; not a mistake, but easy to read

Reply:
replacing map with foreach has meant the rows aren't rendered, is there another step I need to add?